src/loki/reach.cc

```diff
@@ -68,7 +68,8 @@ directed_reach Reach::operator()(const DirectedEdge* edge,
   // seed the expansion with a place to start expanding from
   Clear();
   graph_tile_ptr tile, start_tile = reader.GetGraphTile(edge_id);
-  if ((tile = start_tile) && costing->Allowed(edge, tile, sif::kDisallowSimpleRestriction))
+  if ((tile = start_tile) && costing->Allowed(edge, tile, sif::kDisallowSimpleRestriction) &&
+      !edge->is_shortcut())
```

Comment:
for all of these i would suggest always checking the shortcut before calling the allowed method. basically always do the cheaper thing first as it is more efficient to fail earlier.

```suggestion
  if ((tile = start_tile) && costing->Allowed(edge, tile, sif::kDisallowSimpleRestriction) &&
      !edge->is_shortcut())
```